### PR TITLE
Only check entity attributes for TType_app

### LINKED ISSUE #13525: Internal error when type checking extension method defined in C#
Please provide a succinct description of the issue.

**Repro steps**

Using the latest .NET 7 preview, I cannot compile F# that invokes an extension method which was defined in C#. 
Using my local compiler build, I have the same problem.

Sample:
[TcExprCopyRepro.zip](https://github.com/dotnet/fsharp/files/9141484/TcExprCopyRepro.zip)
[msbuild-binlog.zip](https://github.com/dotnet/fsharp/files/9141506/msbuild-binlog.zip)

Changing the .NET SDK to `6.0.302` (and changing the `<TargetFramework>`) does allow this sample to compile.

**Related information**

The error happens somewhere in https://github.com/dotnet/fsharp/blob/0c1eba0671d0edb86e97618e0a7678a7550befe4/src/Compiler/TypedTree/TypedTreeOps.fs#L877

image

Provide any related information (optional):

* Operating system: Windows
* .NET Runtime kind: .NET 7.0.100-preview.6.22352.1

### PR DESCRIPTION
**Pull request by vzarytovskii**
Fixes #13525

### CODE REVIEW

File: src/Compiler/Checking/AttributeChecking.fs [hunk 1]
```diff
@@ -392,15 +392,18 @@ let CheckILFieldAttributes g (finfo:ILFieldInfo) m =
...
+    if tcref.IsILTycon then 
+        CheckILAttributes g (isByrefLikeTyconRef g m tcref) tcref.ILTyconRawMetadata.CustomAttrs m
     else 
-        CheckFSharpAttributes g x.Attribs m
+        CheckFSharpAttributes g tcref.Attribs m
 
 /// Check the attributes associated with a method, returning warnings and errors as data.
 let CheckMethInfoAttributes g m tyargsOpt (minfo: MethInfo) = 
-    CheckEntityAttributes g minfo.ApparentEnclosingTyconRef m ++ (fun () ->
+    match stripTyEqns g minfo.ApparentEnclosingAppType with
```
Comment (dsyme) [inline src/Compiler/Checking/AttributeChecking.fs:403]: This is OK, but you can use tryTcrefOfAppTy or AppTy active pattern or tryAppTy all of which combine the strip with the test.  In general we can usually avoid a direct pattern match using `TType_*` in most situations, though as long as the appropraite strip is performed they aren't actually wrong.

Comment (vzarytovskii) [reply src/Compiler/Checking/AttributeChecking.fs:403]: Thanks, good to know!

File: tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs [hunk 2]
```diff
@@ -533,4 +533,43 @@ b.text("Hello 2") |> ignore
...
+        namespace CSharp
+        {
+        
+            public interface ITreeNode
+            {
+            }
+        
+            public static class Extensions
+            {
+                public static TNode Copy<TNode>([NotNull] this TNode node, ITreeNode context1 = null) where TNode : ITreeNode =>
```
Comment (dsyme) [inline tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs:557]: I see, so the repro has to have both an attribute and a variable type. Makes sense, and also explains why we haven't encountered this before.

**Review by dsyme — APPROVED** [review]

File: tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs (hunk 2 above)
Comment (vzarytovskii) [reply tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs:557]: Yep

**Pull request merged**